pkg/endpoint/regeneration/owner.go

```diff
@@ -35,6 +35,7 @@ type Owner interface {
 
 	// GetDNSRules creates a fresh copy of DNS rules that can be used when
 	// endpoint is restored on a restart.
+	// No endpoint lock must be held while calling this function.
```

Comment:
nit, the current wording suggests that it's not required to hold the lock, but it's stronger than that. In RFC2119 parlance, we would use "must not" to communicate this clearly.

```suggestion
	// The endpoint lock must not be held while calling this function.
```

Reply:
And to be clear, there are more than one endpoint locks (i.e. `buildMutex`) as Joe mentioned above.

Reply:
Done. I've tried calling it `endpoint.mutex` to be as clear as possible, does that work?